tests/ctests/StatesTest.cpp

```diff
@@ -0,0 +1,444 @@
+    ale::Vec3d(-4.0, -1.0, -4.0),
+    ale::Vec3d(-5.0, -2.0, -3.0),
+    ale::Vec3d(-6.0, -3.0, -2.0),
+    ale::Vec3d(-7.0, -4.0, -1.0)};
+
+  States positionVelocityState(ephemTimes, positions, velocities);
+  vector<ale::State> states = positionVelocityState.getStates();
+  EXPECT_EQ(states.size(), 4);
+  EXPECT_NEAR(states[0].position.x, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].position.y, 1.0, 1e-10);
+  EXPECT_NEAR(states[0].position.z, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.x, -4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.y, -1.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.z, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.x, 7.0, 1e-10);
+  EXPECT_NEAR(states[3].position.y, 4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.z, 1.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.x, -7.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.y, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.z, -1.0, 1e-10);
+}
+
+TEST(StatesTest, ConstructorStates) {
+  std::vector<double> ephemTimes = {0.0, 1.0, 2.0, 3.0};
+  std::vector<ale::Vec3d> positions = {
+    ale::Vec3d(4.0, 1.0, 4.0),
+    ale::Vec3d (5.0, 2.0, 3.0),
+    ale::Vec3d (6.0, 3.0, 2.0),
+    ale::Vec3d (7.0, 4.0, 1.0)};
+
+  std::vector<ale::Vec3d> velocities = {
+    ale::Vec3d(-4.0, -1.0, -4.0),
+    ale::Vec3d(-5.0, -2.0, -3.0),
+    ale::Vec3d(-6.0, -3.0, -2.0),
+    ale::Vec3d(-7.0, -4.0, -1.0)};
+
+  std::vector<ale::State> stateVector;
+  for (int i=0; i < positions.size(); i++) {
+    stateVector.push_back(ale::State(positions[i], velocities[i]));
+  }
+
+  States statesState(ephemTimes, stateVector, 1);
+  vector<ale::State> states = statesState.getStates();
+  EXPECT_EQ(states.size(), 4);
+  EXPECT_NEAR(states[0].position.x, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].position.y, 1.0, 1e-10);
+  EXPECT_NEAR(states[0].position.z, 4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.x, -4.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.y, -1.0, 1e-10);
+  EXPECT_NEAR(states[0].velocity.z, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.x, 7.0, 1e-10);
+  EXPECT_NEAR(states[3].position.y, 4.0, 1e-10);
+  EXPECT_NEAR(states[3].position.z, 1.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.x, -7.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.y, -4.0, 1e-10);
+  EXPECT_NEAR(states[3].velocity.z, -1.0, 1e-10);
+}
+
+class TestState : public ::testing::Test {
+protected:
+  States *states;
+  States *statesNoVelocity;
+
+  // fixtures.h had a lot of overrides before the steup and tearDqwn functions
+  void SetUp() override {
+    states = NULL;
+    statesNoVelocity = NULL;
+
+    //define test data values
+    std::vector<double> ephemTimes = {0.0, 1.0, 2.0, 3.0};
+
+    std::vector<ale::Vec3d> positions = {
+      ale::Vec3d(4, 0, 0),
+      ale::Vec3d(5, 3, 0.064),
+      ale::Vec3d(6, 4, 0.512),
+      ale::Vec3d(7, 3, 1.728)};
+
+    std::vector<ale::Vec3d> velocities = {
+      ale::Vec3d(1, 4, 0),
+      ale::Vec3d(1, 2, 0.48),
+      ale::Vec3d(1, 0, 1.92),
+      ale::Vec3d(1, -2, 4.32)};
```

Comment:
Can you add a comment with what polynomial functions you used for these so that if people need a different time range in the future they know what the values should be.